projects/addon-editor/extensions/file-link/file-link.ts

```diff
@@ -0,0 +1,43 @@
+import {TuiEditorAttachedFile} from '@taiga-ui/addon-editor/interfaces';
+import {Extension} from '@tiptap/core';
+
+declare module '@tiptap/core' {
+    interface Commands<ReturnType> {
+        fileLink: {
+            /**
+             * Comments will be added to the autocomplete.
+             */
+            setFileLink: (preview: TuiEditorAttachedFile) => ReturnType;
+        };
+    }
+}
+
+export const TuiFileLink = Extension.create({
+    name: `fileLink`,
+
+    addCommands() {
+        return {
+            setFileLink:
+                fileLink =>
+                ({state, chain}) => {
+                    const {selection} = state;
+                    const selectedSize = Math.abs(selection.to - selection.from);
+
+                    return (
+                        selectedSize > 0
+                            ? chain().toggleMark(
+                                  `link`,
+                                  {href: fileLink.link},
+                                  {extendEmptyMarkRange: true},
+                              )
+                            : chain().insertContent(
+                                  `<a rel="noopener noreferrer nofollow" href="${fileLink.link}">${fileLink.name}</a><span style="font-size: 15px"> </span>`,
```

Comment:
Maybe add [download](https://www.w3schools.com/tags/att_a_download.asp) attribute, instead of rel?

Reply:
For support non default attribute tiptap requires developers to write custom extensions
<img width="760" alt="image" src="https://user-images.githubusercontent.com/12021443/201914100-5544cfaf-9462-4fa6-a179-23d9a62e1f4c.png">

Reply:
I think we can leave it out for now